File: app/src/main/java/ru/buylist/data/dao/GlobalItemDao.java
package ru.buylist.data.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import ru.buylist.data.entity.GlobalItem;

@Dao
public interface GlobalItemDao {

    @Insert
    void addGlobalItem(GlobalItem globalItem);

    @Delete
    void deleteGlobalItem(GlobalItem globalItem);

    @Update
    void updateGlobalItem(GlobalItem globalItem);

    @Query("SELECT * FROM global_items WHERE name = :name")
    GlobalItem getGlobalItem(String name);
}

File: app/src/main/java/ru/buylist/data/dao/GlobalItemDao.kt
package ru.buylist.data.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import ru.buylist.data.entity.GlobalItem

@Dao
interface GlobalItemDao {

    @Insert
    suspend fun insertGlobalItems(globalItems: List<GlobalItem>)

    @Insert
    suspend fun insertGlobalItem(globalItem: GlobalItem)

    @Update
    suspend fun updateGlobalItem(globalItem: GlobalItem)

    @Query("UPDATE global_items SET color = :color")
    suspend fun updateColor(color: String)

    @Query("SELECT * FROM global_items")
    suspend fun getGlobalItems(): List<GlobalItem>

    @Query("SELECT * FROM global_items WHERE id = :globalItemId")
    suspend fun getGlobalItem(globalItemId: Long): GlobalItem

    @Query("SELECT * FROM global_items")
    fun observeGlobalItems(): LiveData<List<GlobalItem>>

}